runner/builder/builder_lcd.go

```diff
@@ -0,0 +1,87 @@
+package builder
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/mesg-foundation/engine/container"
+	"github.com/mesg-foundation/engine/ext/xos"
+	"github.com/mesg-foundation/engine/hash"
+	"github.com/mesg-foundation/engine/instance"
+	"github.com/mesg-foundation/engine/runner"
+	"github.com/mesg-foundation/engine/service"
+	runnermodule "github.com/mesg-foundation/engine/x/runner"
+)
+
+// LCD is the runner builder returning LCD msg.
+type LCD struct {
+	container    container.Container
+	engineName   string
+	port         string
+	ipfsEndpoint string
+}
+
+// NewLCD returns the runner sdk.
+func NewLCD(container container.Container, engineName, port, ipfsEndpoint string) *LCD {
+	return &LCD{
+		container:    container,
+		engineName:   engineName,
+		port:         port,
+		ipfsEndpoint: ipfsEndpoint,
+	}
+}
+
+// Start start a new runner.
+func (b *LCD) Start(owner sdk.AccAddress, srv *service.Service, env []string) (runnermodule.MsgCreate, error) {
+	instanceEnv := xos.EnvMergeSlices(srv.Configuration.Env, env)
+	envHash := hash.Dump(instanceEnv)
+
+	inst, err := instance.New(srv.Hash, envHash)
+	if err != nil {
+		return runnermodule.MsgCreate{}, err
+	}
+	expRunner, err := runner.New(owner.String(), inst.Hash)
+	if err != nil {
+		return runnermodule.MsgCreate{}, err
+	}
+	expRunnerHash := expRunner.Hash
```

Comment:
All this could be done by calling the querier `/runner/hash`

Reply:
I thought the hash endpoints were just temporary?
But yeah, if so, it will shrink this function as much as nothing: the e2e could directly call the private functions of this package (that should be converted to public ones)

Reply:
done